types/index.d.ts

```diff
@@ -0,0 +1,151 @@
+import * as stream from 'stream';
+import { EventEmitter } from 'events'
+
+
+export type Callback<V, Err = any> = (err: Err, value: V) => void;
+
+export type CodecTransformer = (buffer: Buffer, callback: () => void) => Buffer; // TODO
+
+export interface CodecOptions {
+	deflate: CodecTransformer;
+	snappy: CodecTransformer;
+}
+
+export interface Decoder {
+	on(type: 'metadata', callback: (type: Type.Type) => void): this;
```

Comment:
Does this mean that TypeScript is able to support different callback types for different events?

Reply:
Yes it does!